src/nnet3/nnet-test-utils.cc

```diff
@@ -984,6 +984,12 @@ static void GenerateRandomComponentConfig(std::string *component_type,
       os << "new-column-order=" << buffer.str();
       break;
     }
+    case 21: {
+      *component_type = "PerElementOffsetComponent";
+      os << "dim=" << RandInt(1, 100)
+         << " learning-rate=" << learning_rate << " param-mean=0.0 param-stddev=1.0";
```

Comment:
For testing code, random is good.


On Mon, Nov 2, 2015 at 4:25 PM, hhadian <notifications@github.com> wrote:

> In src/nnet3/nnet-test-utils.cc
> <https://github.com/kaldi-asr/kaldi/pull/307#discussion_r43684673>:
>
> > @@ -984,6 +984,12 @@ static void GenerateRandomComponentConfig(std::string *component_type,
> >        os << "new-column-order=" << buffer.str();
> >        break;
> >      }
> > +    case 21: {
> > +      *component_type = "PerElementOffsetComponent";
> > +      os << "dim=" << RandInt(1, 100)
> > +         << " learning-rate=" << learning_rate << " param-mean=0.0 param-stddev=1.0";
>
> I changed it to randomly select one of the two cases. Is it better to
> revert it to always "param-mean=0.0 param-stddev=1.0" ?
> I also did the rest of modifications.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/307/files#r43684673>.
>